cirq-core/cirq/sim/act_on_args_container_test.py

```diff
@@ -98,6 +103,15 @@ def test_entanglement_causes_join():
     assert args[None] is not args[q0]
 
 
+def test_subcircuit_entanglement_causes_join():
```

Comment:
It would be nice to have a couple of additional tests in this space, e.g.:
- CircuitOp can disentangle qubits
- CircuitOp can (dis-)entangle a subset of the qubits it acts on